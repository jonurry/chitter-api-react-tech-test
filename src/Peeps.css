.peeps {
  align-items: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.peep {
  flex: 1 0 13rem;
  margin: 1rem;
  max-width: 20rem;
  padding: 1rem;
  text-align: center;
}

.peep-text {
  color: #7dbeb8;
  font-family: 'Courier New', Courier, monospace;
  font-size: 1.62rem;
  font-weight: bold;
}

.peep-name-date {
  margin: 0.3rem 0;
}

.peeper {
  color: #a85163;
}

.peep-date {
  color: grey;
  font-family: cursive;
}

button {
  padding: 0.3rem;
}
