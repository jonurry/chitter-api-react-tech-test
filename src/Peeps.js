import React from 'react';
import Peep from './Peep.js';
import './css/Peeps.css';

export default class Peeps extends React.Component {
  render() {
    const { peeps } = this.props;
    return (
      <div className="peeps">
        {peeps.map(peep => (
          <Peep
            key={peep.id}
            peep={peep}
            onDeletePeep={this.props.onDeletePeep}
            currentUser={this.props.currentUser}
          />
        ))}
      </div>
    );
  }
}
