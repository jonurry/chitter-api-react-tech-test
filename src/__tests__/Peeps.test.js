import React from 'react';
import { render } from 'react-testing-library';
import Peeps from '../Peeps';
import Api from '../api.js';

jest.mock('../api');
const api = new Api();
let peeps;
let currentUser;

beforeAll(async () => {
  peeps = await api.getPeeps();
  currentUser = await api.getSession();
});
describe('<Peeps>', () => {
  it('Renders without crashing', () => {
    render(<Peeps peeps={peeps} currentUser={currentUser} />);
  });

  test('Matches snapshot', () => {
    const { container } = render(
      <Peeps peeps={peeps} currentUser={currentUser} />
    );
    expect(container).toMatchSnapshot();
  });
});
