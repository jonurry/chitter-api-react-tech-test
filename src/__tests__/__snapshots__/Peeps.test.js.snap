// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Peeps> Matches snapshot 1`] = `
<div>
  <div
    class="peeps"
  >
    <div
      class="peep"
    >
      <div
        class="peep-text"
        data-testid="peep-text"
      >
        my first peep :)
      </div>
      <div
        class="peep-name-date"
        data-testid="peep-name-date"
      >
        <span
          class="peeper"
        >
          kay
        </span>
         on
         
        <span
          class="peep-date"
        >
          23-06-2018 14:21
        </span>
      </div>
      <button
        data-testid="like-button"
      >
        Likes: 
        1
      </button>
      <button>
        Delete
      </button>
    </div>
    <div
      class="peep"
    >
      <div
        class="peep-text"
        data-testid="peep-text"
      >
        another peep
      </div>
      <div
        class="peep-name-date"
        data-testid="peep-name-date"
      >
        <span
          class="peeper"
        >
          jon
        </span>
         on
         
        <span
          class="peep-date"
        >
          1-07-2018 09:06
        </span>
      </div>
      <button
        data-testid="like-button"
      >
        Likes: 
        2
      </button>
      
    </div>
    <div
      class="peep"
    >
      <div
        class="peep-text"
        data-testid="peep-text"
      >
        and another peep
      </div>
      <div
        class="peep-name-date"
        data-testid="peep-name-date"
      >
        <span
          class="peeper"
        >
          kay
        </span>
         on
         
        <span
          class="peep-date"
        >
          2-07-2018 23:56
        </span>
      </div>
      <button
        data-testid="like-button"
      >
        Likes: 
        2
      </button>
      <button>
        Delete
      </button>
    </div>
  </div>
</div>
`;
